sparkmagic/sparkmagic/magics/remotesparkmagics.py

```diff
@@ -38,6 +38,40 @@ def manage_spark(self, line, local_ns=None):
         Then, create a session. You'll be able to select the session created from the %%spark magic."""
         return self.manage_widget
 
+
+    @magic_arguments()
+    @line_magic
+    @needs_local_scope
+    @argument("-c", "--context", type=str, default=CONTEXT_NAME_SPARK,
+              help="Context to use: '{}' for spark and '{}' for sql queries. "
+                   "Default is '{}'.".format(CONTEXT_NAME_SPARK, CONTEXT_NAME_SQL, CONTEXT_NAME_SPARK))
+    @argument("-o", "--output", type=str, default=None, help="If present, indicated variable will be stored in variable"
+                                                             "of this name in user's local context.")
+    @argument("-m", "--samplemethod", type=str, default=None, help="Sample method for dataframe: either take or sample")
+    @argument("-n", "--maxrows", type=int, default=None, help="Maximum number of rows that will be pulled back "
+                                                                        "from the dataframe on the server for storing")
+    @argument("-r", "--samplefraction", type=float, default=None, help="Sample fraction for sampling from dataframe")
+    @argument("-e", "--coerce", type=str, default=None, help="Whether to automatically coerce the types (default, pass True if being explicit) "
+                                                                        "of the dataframe or not (pass False)")
+    @handle_expected_exceptions
+    def collect(self, line, local_ns=None):
+          args = parse_argstring_or_throw(self.spark, line)
+          coerce = get_coerce_value(args.coerce)
+          if (len(args.command) > 0):
```

Comment:
When would args.command be > 0? Sorry if I'm missing something super obvious.

Reply:
As far as I understand, args.command is the part of the line that does not contain arguments. This part can be empty (for instance if I do ```%spark_collect -o my_dataframe```), or not (```%spark_collect -c sql -o my_df SELECT * from my_table```). 

If it is empty, the join command would throw an error, so this is a way to get around it, but maybe there's a better way...

Reply:
I'm confused... where is args.command coming from? I thought that for you to be able to do that, you needed to define an `@argument` called `command`, and I don't see it there. Am I missing something?